onnxscript/rewriter/pattern.py

```diff
@@ -649,8 +619,11 @@ def pattern(x, shape1, shape2):
         num_outputs = len(node_output_pattern)
         for i, p in enumerate(node_output_pattern):
             assert isinstance(p, NodeOutputPattern)
-            assert p.node_pattern is node_pattern
-            assert p.output_index == i
+            if (p.node_pattern is not node_pattern) or (p.output_index != i):
+                raise NotImplementedError(
+                    "Multi-output pattern not handled by this API. "
+                    "Use other APIs to handle multi-output patterns."
```

Comment:
```suggestion
                    "Use other APIs to handle multi-output patterns for now."
```

Is this temporary?

Reply:
Yes ... for now, they should use Xavier's API to create a rewrite-rule. Eventually, the same API should handle it internally.